changelog/5686.feature.rst

```diff
@@ -0,0 +1 @@
+tmpdir_factory.mktemp now fails given absolute and non-normalized paths.
```

Comment:
```suggestion
``tmpdir_factory.mktemp`` now fails when given absolute and non-normalized paths.
```

Also I think this file should be `5686.improvement.rst` instead of `5686.feature.rst`. 👍

Reply:
fix in latest commit